addOns/exim/src/main/java/org/zaproxy/addon/exim/pcap/PcapUtils.java

```diff
@@ -0,0 +1,234 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2024 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.exim.pcap;
+
+import io.pkts.Pcap;
+import io.pkts.packet.TCPPacket;
+import io.pkts.streams.StreamId;
+import io.pkts.streams.TcpStream;
+import io.pkts.streams.impl.TcpStreamHandler;
+import io.pkts.streams.impl.TransportStreamId;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.Map;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.parosproxy.paros.network.HttpHeader;
+import org.parosproxy.paros.network.HttpMalformedHeaderException;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpRequestHeader;
+import org.parosproxy.paros.network.HttpResponseHeader;
+import org.zaproxy.zap.network.HttpRequestBody;
+import org.zaproxy.zap.network.HttpResponseBody;
+
+public final class PcapUtils {
+
+    private static final Logger LOGGER = LogManager.getLogger(PcapUtils.class);
+
+    private static final String DOUBLE_CRLF = HttpHeader.CRLF + HttpHeader.CRLF;
+
+    private static Map<StreamId, TcpStream> extractTcpStreams(File pcapFile) throws IOException {
+        TcpStreamHandler streamHandler = new TcpStreamHandler();
+
+        final Pcap pcap = Pcap.openStream(pcapFile);
+        pcap.loop(streamHandler);
+        pcap.close();
+
+        return streamHandler.getStreams();
+    }
+
+    public static Collection<TcpStream> extractHttpStreams(File pcapFile) throws IOException {
```

Comment:
Any reason for Collection instead of a list?

Reply:
No particular reason, I only use the data structure to iterate over it so I went with the highest abstraction. I'll move it to `List<TcpStream>` if you think it's best.